# Utilising large and diverse bacterial genome datasets to improve the identification of *Streptococcus pneumoniae* via PCR-based diagnostics

This document contains the code used to analyse the data for the PCR manuscript.
The PCR manuscript - working title "Utilising large and diverse bacterial genome datasets to improve the identification of *Streptococcus pneumoniae* via PCR-based diagnostics" - contains four Figures and three Tables in the main text. The Supplementary Materials contain seven Figures and six Tables. 

## (ECCMID) Abstract
### Background
Accurate identification of *Streptococcus pneumoniae* (pneumococcus) is crucial for clinical microbiology, surveillance and epidemiology, but can be complicated by the misidentification of non-pneumococcal *Streptococcus* (NPS) species. The aims of this study were to use genome datasets to assess the *in silico* accuracy of current qPCR/PCR targets, and then develop and validate *in vitro* assays for the most promising target: the ‘Xisco’ gene (TIGR4 genome locus SP_1992). 

### Methods
Three Xisco qPCR primer/probe sets were designed and compared to six published qPCR primer/probe sets for *lytA, piaB, ply, psaA,* Spn9802 and SP2020, and one published PCR primer set for Xisco, using the *in silico* PCR tool in PubMLST (https://bigsdb.readthedocs.io/en/latest/data_analysis/in_silico_pcr.html). Overall, 7,547 pneumococcal genomes, and 1,825 genomes that represented 55 NPS species, were analysed *in silico*. Putative qPCR/PCR amplicon sequences were aligned and up to four nucleotide mismatches per primer or probe were allowed to be considered a positive result. The three Xisco qPCR primer/probe sets and one published primer/probe set each for *lytA* and *piaB* were tested *in vitro* using 246 pneumococci from the 2022 UK surveillance collection and 277 isolates of 21 different NPS species.

### Results
Among all 7,547 pneumococcal genomes, the Xisco primer/probe sets correctly detected virtually all true positives *in silico* (99.9% of true positives, 95% CI 99.8-100%) and Xisco was never detected among 1,825 NPS genomes. In contrast, all other non-Xisco qPCR/PCR targets detected 96.8-100% of pneumococci *in silico*, and produced *in silico* false positives for between 1 and 82 NPS genomes (*Streptococcus pseudopneumoniae* or *Streptococcus mitis*). Therefore, *in vitro* qPCR assays were developed for Xisco, with a resulting overall sensitivity of 100% (246/246, 95% CI 98.5-100%) among pneumococci and overall specificity of 100% (277/277, 95% CI 98.7-100%) among NPS isolates, which was comparable or superior to the tested *lytA* and *piaB in vitro* assays.

### Conclusions
Large-scale *in silico* analyses of diverse genome datasets are a cost-effective method to investigate the putative accuracy of qPCR primers/probes prior to the more expensive development of *in vitro* laboratory assays.
